content/articles/how-to-build-a-ussd-application-with-django/index.md

```diff
@@ -0,0 +1,513 @@
+            response += "1. Nigerian Naira (NGN)\n"
+            response += "2. US Dollars (USD)\n"
+            response += "3. Euro (EUR)\n"
+        else:
+            response = "END. Invalid input. Please try again"
+            return HttpResponse(response)
+    elif input[0] == "2":
+        possible_input = ['1', '2', '3', '4']
+        
+        if input[1] in possible_input:
+            response = "CON Choose the currency\n" # gotta find better word for this
+            response += "1. Nigerian Naira (NGN)\n"
+            response += "2. US Dollars (USD)\n"
+            response += "3. Euro (EUR)\n"
+            return HttpResponse(response)
+        else:
+            response = "END. Invalid input. Please try again"
+            return HttpResponse(response)
+```
+
+## Explanation
+`len(input) == 2` simply means the user has gone through two levels of menus i.e, they have supplied input twice, and I have split those inputs into elements of a list.
+I then checked if the first element (first user input) in the list is "1" if it is it means the user wants to check the price of a cryptocurrency, I then went further to check if the second input is any of the possible input to be entered, i.e 1 to 4, if it is, it means they have chosen a cryptocurrency whose price they want to check, I then proceed to ask them to choose the fiat currency in which they want to know how much their chosen cryptocurrency costs.
+Same thing if the first element in the list is "2", it means the user wants to know the exchange rate between two currencies. I then proceed to check if the second element in the list which is the number corresponding to the base currency is valid, if it is, it means they have chosen their base currency and I can now ask them to choose a number corresponding to the currency they want to use as the "to currency".
+
+As explained above, the user input will be appended to the text parameter. For example, if a user wants to know the price of bitcoin in US Dollars, the value of text at this level will be "1 *1 *2", and input will be ["1", "1", "2"].
+
+Similar to what I did in steps A and B, I will also check the elements in the list to know what the user inputs are. So under `elif len(input) == 2`, on the same indentation level, enter the code below
+```
+elif len(input) == 3:
+    if input[0] == "1":
+        # BITCOIN
+        if input[1] == "1":
+            if input[2] == "1":
+                result = get_cryptocurrency_price('bitcoin', 'ngn')
+                Notice how I started the response with the "END" keyword to indicate that I want that session to end after users have been shown the price of the cryptocurrency they choose.
+                return HttpResponse(response)
+            elif input[2] == "2":
+                result = get_cryptocurrency_price('bitcoin', 'usd')
+                response = f"END Current price of Bitcoin in US Dollars is {result['bitcoin']['usd']}"
```

Comment:
Break down this entire code snippet into several smaller snippets. And, explain each of them